### DISCUSSION TITLE
Bundle Size

### DISCUSSION THREAD

**Discussion by davecarlson** [onesine/react-tailwindcss-datepicker#160] [General]
I've not raised as an issue as it feels more like a discussion.

I've been analysing our next.js web app (compliles fully to static as all the data comes from an external API ), I notice the this moduel is quick high up on the size chart. Is this something others have seen? Actually only using it in 1 component, then using that component in 6 of the ~ 80 pages in the app:

The module is the middle large box on the left hand side :)

![Screenshot 2023-07-04 at 19 45 24 Large](https://github.com/onesine/react-tailwindcss-datepicker/assets/299702/5725b8b8-7a81-49cb-ba9a-f217e1000b42)

**Comment by davecarlson** [comment]
A close up on the stats 
<img width="1305" alt="Screenshot 2023-07-04 at 19 49 46" src="https://github.com/onesine/react-tailwindcss-datepicker/assets/299702/da579815-3d62-426f-b0fb-9b0d7952c839">

**Comment by simpros** [comment]
+1 this lib is way to big

**Comment by CremboC** [comment]
According to [Bundlephobia](https://bundlephobia.com/package/react-tailwindcss-datepicker) a change was made between 1.4.1 and 1.4.2 that massively increased the size the lib. I have a strong suspicion it's [this commit](https://github.com/onesine/react-tailwindcss-datepicker/pull/56/files). I wouldn't be surprised it broke tree shaking or something, although I'm not familiar enough with JS world to be certain.

**Reply by davecarlson** [reply]
I would suspect so!

**Comment by willashley23** [comment]
Any update here? I'm seeing the same thing in my Next.js app. I tried downgrading to `1.4.1` which nominally reduces the bundle size, but it seems to bring with it a ton of dayjs locale imports that I think are being used in that version, which actually makes the overall bundle even bigger

<img width="1064" alt="image" src="https://github.com/onesine/react-tailwindcss-datepicker/assets/9400869/84c348e4-e395-4aca-9718-ab2dcd07542c">
